test/e2e/nodeportlocal_test.go

```diff
@@ -167,12 +169,30 @@ func updateNPLPortRangeInConfigmap(t *testing.T, data *TestData, newStartPort, n
 	}
 }
 
-func validatePortInRange(t *testing.T, nplAnnotations []k8s.NPLAnnotation, start, end int) {
+func validatePortsInAnnotation(t *testing.T, r *require.Assertions, nplAnnotations []k8s.NPLAnnotation, start, end int, targetPorts sets.Int) {
+	require.Equal(t, len(targetPorts), len(nplAnnotations))
 	for i := range nplAnnotations {
-		if nplAnnotations[i].NodePort > end || nplAnnotations[i].NodePort < start {
-			t.Fatalf("Node port %d not in range: %d - %d", nplAnnotations[i].NodePort, start, end)
+		podPort := nplAnnotations[i].PodPort
+		r.True(targetPorts.Has(podPort), "Port %d in Pod annotation not found in set target ports :%v from Services", podPort, targetPorts)
+		targetPorts.Delete(podPort)
+
+		nodePort := nplAnnotations[i].NodePort
+		if nodePort > end || nodePort < start {
+			t.Fatalf("Node port %d not in range: %d - %d", nodePort, start, end)
 		}
 	}
+	r.Emptyf(targetPorts, "Target ports %v not found in Pod annotation", targetPorts)
+}
+
+// createNginxPodWithPort creates an nginx Pod in the test namespace with configurable container port.
+func createNginxPodWithPort(name, nodeName string, port int32) error {
```

Comment:
I think we should stick with a method on the test data object for consistency:
```go
func (data *testData) createNginxPodWithPort(...)
```

Reply:
Done